### PR TITLE
Make a few improvements to the gallery config.

### PR DESCRIPTION
**Pull request by Cadair**
This mainly just changes the URL to be `gallery` rather than `auto_examples`.

### CODE REVIEW

**Comment by Cadair** [comment]
@sunpy/sunpy-developers review please!

@Alex-Ian-Hamilton This means you don't need to start the file with `plot_`

**Comment by nabobalis** [comment]
Looks fine to me.

File: doc/source/conf.py [hunk 1]
```diff
@@ -80,7 +80,7 @@
 # This does not *have* to match the package name, but typically does
 project = u'SunPy'
 author = u'The SunPy Community'
-copyright = u'2013, ' + author
+copyright = u'2016, ' + author
```
Comment (dpshelio) [inline doc/source/conf.py]: Good one! we've got the :copyright: notice outdated for three years! cannot we use `datetime.datetime.now().year`?

File: doc/source/conf.py [hunk 2]
```diff
@@ -179,7 +179,15 @@
...
-    # path where to save gallery generated examples
-    'gallery_dirs'  : 'auto_examples'}
+    # execute all examples except those that start with "skip_"
+    'filename_pattern': '^((?!skip_).)*$',
+    # path to the examples scripts
+    'examples_dirs': '..{}..{}examples'.format(os.sep, os.sep),
+    'gallery_dirs': 'gallery',
+    'reference_url': {
+        'sunpy': None,
+        'astropy': 'http://docs.astropy.org/en/latest/',
```
Comment (dpshelio) [inline doc/source/conf.py]: shouldn't be `stable` instead of `latest`?

**Comment by Cadair** [comment]
@dpshelio approve?

**Comment by Cadair** [comment]
I need this for a proposal I am writing so I am going to merge it.

**Pull request merged**